base/stats.go

```diff
@@ -0,0 +1,747 @@
+package base
+
+import (
+	"expvar"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	StatViewFormat = "%s.%s"
+)
+
+type SgwStats struct {
+	GlobalStats     GlobalStat          `json:"global"`
+	DbStats         map[string]*DbStats `json:"per_db"`
+	ReplicatorStats *ReplicatorStats    `json:"per_replication"`
+
+	initDBStats               sync.Once
```

Comment:
In general it's preferable to avoid sync.Once unless absolutely needed.  In this case it seems like we should just have a constructor for SgwStats that initializes DbStats and ReplicatorStats, and maybe stick omitempty on ReplicatorStats?

Reply:
Makes sense for dbstats. Not sure about replicatorStats. I feel like paying for the sync may be better than initializing the descriptions and map when it might not be needed?

Reply:
Initializing the descriptions doesn't seem particularly expensive, and ReplicatorStats is just a map, I think initializing an empty map (with omitempty on the json annotation) is worth it for the simplified code.

Reply:
Fair enough